### DISCUSSION TITLE
glsp-vscode-client and node server

### DISCUSSION THREAD

**Discussion by csamarchi** [eclipse-glsp/glsp#668] [General]
Hello, I've been trying to get the [glsp-vscode-integration](https://github.com/eclipse-glsp/glsp-vscode-integration) workflow example working with the [glsp-node-server](https://github.com/eclipse-glsp/glsp-server-node). I made a few adjustments inside the `workflow-extension.ts` file and it seems to be connecting to the server. See below screenshot 
![Screen Shot 2022-05-31 at 4 12 10 PM](https://user-images.githubusercontent.com/42501494/171285310-2a908b6a-18db-4210-8f45-745d6cce95d3.png)

However when I go to run the Worklfow GLSP Extension Example as it says to do, it is empty. Not seeing anything. Not seeing any error logs in the debug terminal either. 
![Screen Shot 2022-05-31 at 4 16 58 PM](https://user-images.githubusercontent.com/42501494/171285931-27486775-ea58-4f5c-a043-2657d310fe57.png)

These are the steps I've ran to produce:
1. `yarn install `and `yarn build` inside the client directory
2. `yarn install` and `yarn build` and `yarn start:server` inside the node server directory
3. Open VS Code (for client) and run Workflow GLSP Example Extension (this works for me when running with the java server but not the node one)

Can anybody help me with this? Any debugging ideas? Anybody tried getting the glsp vscode extension working with the glsp node server?

**Comment by tortmayr** [comment]
Hi @csamarchi,

> Not seeing anything. Not seeing any error logs in the debug terminal either.

The GLSP diagram editor is running in an isolated webview. Log messages of this webview are not logged in the debug view.
However you can use the `Developer Tools` (Help -> Toggle Developer Tools)  inside of the runtime VS Code instance. Then you can check for errors in the `Console` tab.

>Can anybody help me with this? Any debugging ideas? Anybody tried getting the glsp vscode extension working with the glsp node server?

We recently added a [project template](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-vscode) to the GLSP examples repository that demonstrates how to use the VS Code integration with a node GLSP server. Using this template as baseline you should be able to adapt it for the workflow example with only minimal changes.

I can only make an educated guess at this point but in general your approach and the changes you posted look fine to me. Could be a dependency resolution issue. In the project template we use [webpack](https://github.com/eclipse-glsp/glsp-examples/blob/master/project-templates/node-json-vscode/glsp-server/webpack.config.js) to bundle the server sources into a single  minified file. Maybe you should use the same approach for the workflow example and then update the `executable` path so that it points to the packed .js file.
